Sparkle/SUBinaryDeltaCreate.m

```diff
@@ -138,11 +220,25 @@ int createBinaryDelta(NSString *source, NSString *destination, NSString *patchFi
         }
 
         NSDictionary *info = infoForFile(ent);
+        if (!info) {
+            fprintf(stderr, "Failed to retrieve info for file %s", ent->fts_path);
+            return 1;
+        }
         originalTreeState[key] = info;
+        
+        if (aclExists(ent)) {
+            fprintf(stderr, "Diffing ACLs are not supported. Detected ACL in before-tree on file %s", ent->fts_path);
+            return 1;
+        }
     }
     fts_close(fts);
+    
+    NSString *beforeHash = hashOfTreeWithVersion(source, majorVersion);
```

Comment:
If I understand correctly, the beforeHash will include permissions. However, existing permissions aren't necessary to patch to be correct — they could simply be replaced with new values.

For that reason maybe it's best to ignore them when creating beforeHash?

Or even don't include them in the hash at all (which would simplify some logic). If you want to check them at all, then add a new, separate "afterPermissionsHash" that's built only from files permissions and is only checked against the final state.